industry_benchmarks/utils/tests/test_data_gathering.py

```diff
@@ -156,4 +200,190 @@ def test_get_sasa_diff(cmet_ligand_network):
+        """
+        runner = CliRunner()
+        result = runner.invoke(
+            gather_data,
+            [
+                "--input_alchemical_network",
+                str(bace_full_results / "alchemicalNetwork" / "alchemical_network.json"),
+                "--output_dir",
+                str(tmpdir / "full_test"),
+                "--results-folder",
+                str(bace_full_results / "results_0"),
+                "--results-folder",
+                str(bace_full_results / "results_1"),
+                "--results-folder",
+                str(bace_full_results / "results_2")
+            ]
+        )
+        assert result.exit_code == 0
+        # make sure the missing png file warnings are printed
+        assert "Can't find cleaned results file: forward_reverse_convergence.png" in result.stdout
+        # make sure we have a zip folder
+        archive: pathlib.Path = tmpdir / "full_test.zip"
+        assert archive.exists()
+        # make sure we have the folder made into the archive
+        result_folder = tmpdir / "full_test"
+        assert result_folder.exists()
+        # load the results JSON
+        result_json = json.load(
+            (result_folder / "all_network_properties.json").open(mode="r"),
+            cls=JSON_HANDLER.decoder
+        )
+        # check for the expected keys
+        assert "Network_map" in result_json
+        assert "transformation_scores" in result_json
+        assert "ligand_scores" in result_json
+        assert "DDG_estimates" in result_json
+
+        # do some basic checks
+        # check for 2 ligands and a single edge
+        assert len(result_json["Network_map"]["nodes"]) == 2
+        assert len(result_json["Network_map"]["edge"]) == 1
+        # check the single edge has a score
+        assert len(result_json["transformation_scores"]) == 1
+        assert "edge_spiro2_spiro1" in result_json["transformation_scores"]
+        # check the ligand scores
+        assert len(result_json["ligand_scores"]) == 2
+        assert "ligand_spiro2" in result_json["ligand_scores"]
+        assert "ligand_spiro1" in result_json["ligand_scores"]
+
+        # check the edge estimates
+        assert len(result_json["DDG_estimates"]) == 6
+        for repeat in range(3):
+            for phase in ["solvent", "complex"]:
+                edge_name = f"{phase}_spiro2_spiro1_repeat_{repeat}"
+                assert edge_name in result_json["DDG_estimates"]
+                # make sure the edge folder was created
+                edge_folder = result_folder / edge_name
+                assert edge_folder.exists()
+                # check the files were moved to the directory
+                for f_name in [
+                    "structural_analysis_data.npz",
+                    "energy_replica_state.npz",
+                    "simulation_real_time_analysis.yaml",
+                    "info.yaml"
+                ]:
+                    assert (edge_folder / f_name).exists()
```

Comment:
Should we add a test for the case where we have two alchemical network files, the "old" one (where some simulation had failed) and the new one from the fix_network script?

Reply:
Great idea, added some variations on this to test failures and normal running with a fixed network.